### PR TITLE
I111 iddindex

### PR DESCRIPTION
**Pull request by jamiebull1**
Creating pull request for code review/comments

### CODE REVIEW

File: eppy/EPlusInterfaceFunctions/iddindex.py [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+    return ref2namesdct
+    
+def ref2names2commdct(ref2names, commdct):
+    """embed ref2names into commdct"""
+    for comm in commdct:
+        for cdct in comm:
+            try:
+                refs = cdct['object-list'][0]
+                pointsto = ref2names[refs]
+                # pointsto = set(pointsto)
```
Comment (jamiebull1) [inline eppy/EPlusInterfaceFunctions/iddindex.py]: Was this meant to be used?

File: eppy/EPlusInterfaceFunctions/parse_idd.py [hunk 2]
```diff
@@ -78,13 +79,9 @@ def removeblanklines(astr):
...
     lines = [line for line in lines if line.strip() != ""]
     return "\n".join(lines)
-
-@decorator
-def embeddgroupdata(extract_func, fname, debug=False):
-    """insert group info into extracted idd"""
     
-    # copied the reading of fname from extractidddata - start
-    # removed decode (the decode will happen in extractidddata)
+def _readfname(fname):
```
Comment (jamiebull1) [inline eppy/EPlusInterfaceFunctions/parse_idd.py]: Why copied from `extractidddata` and not imported?

File: eppy/EPlusInterfaceFunctions/parse_idd.py [hunk 3]
```diff
@@ -95,7 +92,34 @@ def embeddgroupdata(extract_func, fname, debug=False):
             astr = fname.read()
         else:
             astr = mylib2.readfile(fname)
-    # copied the reading of fname from extractidddata- end
+    return astr
+            
+@decorator
+def make_idd_index(extract_func, fname, debug):
+    """generate tie iddindex"""
```
Comment (jamiebull1) [inline eppy/EPlusInterfaceFunctions/parse_idd.py]: Typo?

File: eppy/EPlusInterfaceFunctions/parse_idd.py [hunk 4]
```diff
@@ -95,7 +92,34 @@ def embeddgroupdata(extract_func, fname, debug=False):
...
+    
+    name2refs = iddindex.makename2refdct(commdct)
+    ref2namesdct = iddindex.makeref2namesdct(name2refs)
+    idd_index = dict(name2refs=name2refs, ref2names=ref2namesdct)
+    commdct = iddindex.ref2names2commdct(ref2namesdct, commdct)
+    
+    return blocklst, commlst, commdct, idd_index
+
+@decorator
+def embeddgroupdata(extract_func, fname, debug):
```
Comment (jamiebull1) [inline eppy/EPlusInterfaceFunctions/parse_idd.py]: Typo - should this be `embedgroupdata`?

File: eppy/EPlusInterfaceFunctions/parse_idd.py [hunk 5]
```diff
@@ -111,15 +135,21 @@ def embeddgroupdata(extract_func, fname, debug=False):
...
     see what the change is
-    - 
-    Does not integrate group data into the results
+    -
+    this code is from 2004.
+    it works.
+    I am trying not to change it (until I rewrite the whole thing) 
+    to add functionality to it, I am using decorators
+    So Now
+    Does not integrate group data into the results (@embeddgroupdata does it)
```
Comment (jamiebull1) [inline eppy/EPlusInterfaceFunctions/parse_idd.py]: Same typo as above?

File: eppy/EPlusInterfaceFunctions/parse_idd.py [hunk 6]
```diff
@@ -111,15 +135,21 @@ def embeddgroupdata(extract_func, fname, debug=False):
...
     extracts all the needed information out of the idd file
     if debug is True,  it generates a series of text files.
     Each text file is incrementally different. You can do a diff
     see what the change is
-    - 
-    Does not integrate group data into the results
+    -
+    this code is from 2004.
+    it works.
+    I am trying not to change it (until I rewrite the whole thing)
```
Comment (jamiebull1) [inline eppy/EPlusInterfaceFunctions/parse_idd.py]: Seems wise!

File: eppy/tests/EPlusInterfaceFunctions_tests/test_iddindex.py [hunk 7]
```diff
@@ -0,0 +1,1934 @@
...
+            # but is pointing to the set in ref2names
+            for item in r_item:
+                try:
+                    reference = item['object-list'][0]
+                    pointsto = item['pointsto']
+                    assert id(ref2names[reference]) == id(pointsto)
+                except KeyError as e:
+                    continue
+    
+# full commdct for the following idfobjs
```
Comment (jamiebull1) [inline eppy/tests/EPlusInterfaceFunctions_tests/test_iddindex.py]: Is all this needed?

**Comment by jamiebull1** [comment]
Yep, this looks good. The `thermal_properties` tests are running a lot quicker now - 0.02 seconds rather than around 2 seconds so a big gain!

File: eppy/EPlusInterfaceFunctions/iddindex.py (hunk 1 above)
Comment (santoshphilip) [inline eppy/EPlusInterfaceFunctions/iddindex.py]: will delete

File: eppy/EPlusInterfaceFunctions/parse_idd.py (hunk 2 above)
Comment (santoshphilip) [inline eppy/EPlusInterfaceFunctions/parse_idd.py]: It is is a snippet copied from the function `extractidddata` which is further below in the code
I have changed the documentation to clarify that

File: eppy/EPlusInterfaceFunctions/parse_idd.py (hunk 3 above)
Comment (santoshphilip) [inline eppy/EPlusInterfaceFunctions/parse_idd.py]: fixed.

File: eppy/EPlusInterfaceFunctions/parse_idd.py (hunk 4 above)
Comment (santoshphilip) [inline eppy/EPlusInterfaceFunctions/parse_idd.py]: yes. fixed.

File: eppy/tests/EPlusInterfaceFunctions_tests/test_iddindex.py (hunk 7 above)
Comment (santoshphilip) [inline eppy/tests/EPlusInterfaceFunctions_tests/test_iddindex.py]: removed

**Comment by santoshphilip** [comment]
@jamiebull1 , I had done this merge `i111_iddindex` into `develop` on my laptop.
Forgot to do `git push origin develop`

I know that you are branching from `develop` for your work on `six` and `improved coverage`. There is a chance that you may get difficult merge conflicts with those branches once I do the push that I forgot.

so I pushed it as `develop_2016-07-22`
When you get a moment see if this will cause merge conflicts on the branches you are working on.
If it looks OK, I'll push it as `develop`

**Comment by jamiebull1** [comment]
Ok, I'll take a look when I get a moment. By the way, any idea when you can get to putting the updated master onto PyPI?  I'm doing some stuff with geometry - have a look at geomeppy on my channel - and it won't work with python 3 till pip install works again.

**Comment by santoshphilip** [comment]
I'll do it today

**Comment by jamiebull1** [comment]
Awesome, thanks. I'm looking at the merges now. Looks clean so far.

**Comment by jamiebull1** [comment]
Doesn't seem to be any problem there. Looks ok to push to `develop` to me.

**Comment by santoshphilip** [comment]
Will do the push into `develop`

tag 0.5.3 (latest master) has been published on pypi

**Pull request merged**